File: docs/websocket-transcript-example.js
// Example of using the WebSocket transcript streaming endpoint
// Replace with your actual bot ID and API key
const BOT_ID = 'bot_xxxxxxxxxxx';
const API_KEY = 'your_api_key_here';

const socket = new WebSocket(`wss://app.attendee.dev/ws/bots/${BOT_ID}/transcript/?token=${API_KEY}`);

// Handle connection open
socket.addEventListener('open', event => {
    console.log('WebSocket connection established');
});

// Handle messages
socket.addEventListener('message', event => {
    const data = JSON.parse(event.data);
    
    if (data.type === 'initial_transcript') {
        // Handle initial transcript data
        console.log('Initial transcript received:', data.utterances.length, 'utterances');
        
        // Process each utterance in the initial transcript
        data.utterances.forEach(utterance => {
            displayUtterance(utterance);
        });
    } 
    else if (data.type === 'new_utterance') {
        // Handle new utterance that came in real-time
        console.log('New utterance received:', data.utterance);
        displayUtterance(data.utterance);
    }
});

// Handle errors
socket.addEventListener('error', event => {
    console.error('WebSocket error:', event);
});

// Handle connection close
socket.addEventListener('close', event => {
    console.log('WebSocket connection closed with code:', event.code);
    
    // You might want to implement reconnection logic here
    if (event.code === 4003) {
        console.error('Authentication failed. Check your API key.');
    }
});

// Example function to display an utterance in the UI
function displayUtterance(utterance) {
    // Example implementation - replace with your actual UI logic
    const transcriptContainer = document.getElementById('transcript-container');
    
    const utteranceElement = document.createElement('div');
    utteranceElement.className = 'utterance';
    
    const speakerElement = document.createElement('div');
    speakerElement.className = 'speaker';
    speakerElement.textContent = utterance.speaker_name || 'Unknown Speaker';
    
    const textElement = document.createElement('div');
    textElement.className = 'text';
    textElement.textContent = utterance.transcription.transcript;
    
    utteranceElement.appendChild(speakerElement);
    utteranceElement.appendChild(textElement);
    
    transcriptContainer.appendChild(utteranceElement);
    
    // Auto-scroll to the bottom
    transcriptContainer.scrollTop = transcriptContainer.scrollHeight;
}

// Example of how to close the connection when done
function closeConnection() {
    socket.close();
}

// Example of how to reconnect if needed
function reconnect() {
    if (socket.readyState === WebSocket.CLOSED) {
        socket = new WebSocket(`wss://app.attendee.dev/ws/bots/${BOT_ID}/transcript/?token=${API_KEY}`);
        // Re-add all event listeners
    }
} 

File: docs/websocket-transcript-example.py
#!/usr/bin/env python3
"""
Example Python client for the WebSocket transcript streaming endpoint.
Requires websockets package: pip install websockets
"""

import asyncio
import json
import websockets

# Replace with your actual bot ID and API key
BOT_ID = 'bot_xxxxxxxxxxx'
API_KEY = 'your_api_key_here'
BASE_URL = 'wss://app.attendee.dev'

async def connect_to_transcript_websocket():
    """Connect to the transcript WebSocket endpoint and handle messages."""
    url = f"{BASE_URL}/ws/bots/{BOT_ID}/transcript/?token={API_KEY}"
    
    try:
        async with websockets.connect(url) as websocket:
            print("WebSocket connection established")
            
            while True:
                # Wait for messages
                message = await websocket.recv()
                data = json.loads(message)
                
                if data.get("type") == "initial_transcript":
                    # Handle initial transcript data
                    utterances = data.get("utterances", [])
                    print(f"Initial transcript received: {len(utterances)} utterances")
                    
                    # Process each utterance in the initial transcript
                    for utterance in utterances:
                        display_utterance(utterance)
                
                elif data.get("type") == "new_utterance":
                    # Handle new utterance that came in real-time
                    utterance = data.get("utterance")
                    print(f"New utterance received from {utterance.get('speaker_name')}")
                    display_utterance(utterance)
    
    except websockets.exceptions.InvalidStatusCode as e:
        if e.status_code == 4003:
            print("Authentication failed. Check your API key.")
        else:
            print(f"Connection failed with status code: {e.status_code}")
    
    except websockets.exceptions.ConnectionClosed as e:
        print(f"Connection closed with code: {e.code}, reason: {e.reason}")
    
    except Exception as e:
        print(f"Error: {e}")

def display_utterance(utterance):
    """Display an utterance in the console."""
    speaker = utterance.get("speaker_name", "Unknown Speaker")
    text = utterance.get("transcription", {}).get("transcript", "")
    timestamp_ms = utterance.get("timestamp_ms", 0)
    
    # Convert timestamp to seconds
    timestamp_sec = timestamp_ms / 1000
    minutes = int(timestamp_sec // 60)
    seconds = int(timestamp_sec % 60)
    timestamp_formatted = f"{minutes:02}:{seconds:02}"
    
    print(f"[{timestamp_formatted}] {speaker}: {text}")

if __name__ == "__main__":
    asyncio.run(connect_to_transcript_websocket()) 